catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SeedCommand.java

```diff
@@ -0,0 +1,195 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.catalog;
+
+import static java.util.AbstractMap.SimpleEntry;
+import static java.util.Map.Entry;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+import static ddf.catalog.data.Metacard.ANY_TEXT;
+import static ddf.catalog.filter.FilterDelegate.WILDCARD_CHAR;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.fusesource.jansi.Ansi;
+import org.geotools.filter.text.cql2.CQL;
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.ResourceRequestById;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+
+@Service
+@Command(scope = CatalogCommands.NAMESPACE, name = "seed", description = "Seeds the local metacard "
+        + "and product caches from the enterprise or from specific federated sources.", detailedDescription =
+        "This command will query the enterprise or the specified sources for metacards in increments "
+                + "the size of the `--product-limit` argument (default 20) until that number of product downloads "
+                + "have started or until there are no more results. Local products will not be added to the cache. "
+                + "This command will not re-download products that are up-to-date in the cache, so subsequent runs "
+                + "of the command will attempt to cache metacards and products that have not already been cached.\n\n"
+                + "Note that this command will begin the product downloads and some may still be in progress "
```

Comment:
This sentence isn't very clear for describing the async nature of the command. Suggest something like...
`Note: this command will trigger product downloads in the background. Those downloads may continue after control is returned to the console.`